models/articles.model.js

```diff
@@ -117,47 +117,88 @@ exports.fetchCommentsForArticle = (
   });
 };
 
+exports.postComment = (reqBody, articleId) => {
+  if (!reqBody.username || !reqBody.body) {
+    return Promise.reject({ code: "22P02" });
+  }
 
+  const checkArticle = checkArticleExists(articleId);
+  const queryPromise = db
+    .query(`SELECT * FROM articles;`)
+    .then(({ rows: articleRows }) => {
+      reqBody.created_at = Date.now();
+      reqBody.article_id = articleId;
+      reqBody.author = reqBody.username;
+
+      const articleIdLookup = createRef(articleRows, "title", "article_id");
+      const formattedCommentData = formatComments([reqBody], articleIdLookup);
+      const insertCommentsQueryStr = format(
+        "INSERT INTO comments (body, author, article_id, votes, created_at) VALUES %L RETURNING *;",
+        formattedCommentData.map(
+          ({ body, author, article_id, votes = 0, created_at }) => [
+            body,
+            author,
+            article_id,
+            votes,
+            created_at,
+          ]
+        )
+      );
+
+      return db.query(insertCommentsQueryStr).then((result) => {
+        return result.rows[0];
+      });
+    });
 
+  return Promise.all([checkArticle, queryPromise]).then((result) => {
+    return result[1];
+  });
+};
 
-
-
-
-exports.postComment = (reqBody, articleId) => {
-  if (!reqBody.username || !reqBody.body) {
-    return Promise.reject({code: '22P02'})
+exports.updateArticle = (articleId, propertiesToUpdate) => {
+  if (Object.keys(propertiesToUpdate).length === 0) {
+    return Promise.reject({ code: "22P02" });
```

Comment:
This is not what we want to be doing. 22P02 would be a PSQL error that we make use of. We would more typically directly use the http error code/msg we want to send as our response.